### ISSUE TITLE
running PLElog.py problem

### ISSUE THREAD

**Issue by xiaooomd** [YangLin-George/PLELog#21]
Hello，Dr.Yang, I want to reproduce the PLElog algorithm for processing HDFS. According to the description of preparation in readme.md, I just need to place the hdfs dataset and glove.6B.300d.txt under the datasets folder to run PLElog.py. 
I downloaded the hdfs dataset and exception tag file, followed the instructions to place the file and run PLElog.py in pycharm. The program ran for 4-5 hours without any results. Am I running the wrong steps? If I run the wrong steps, please point out my mistake.

The following three pictures are screenshots of the files I configured according to the requirements and running the program. It has been running for 4-5 hours, and there are no errors reported in this interface.
![datasets](https://github.com/LeonYang95/PLELog/assets/51357799/69ad7d8f-8c9b-4d01-ad36-38e3e59f724b)
![datasets1](https://github.com/LeonYang95/PLELog/assets/51357799/c390147f-b71a-4fb4-923e-4db1bae3f2d3)
![PLElog](https://github.com/LeonYang95/PLELog/assets/51357799/c600caeb-4278-4cb1-9126-8e11c052f189)

**Comment by LeonYang95** [comment]
Theoretically, the _load_raw_log_seqs method in HDFSLoader.py(Line 149) will read the HDFS.log file to produce log sequences when the persistence files do not exist. However, it should print "_Start extract log sequences from HDFS raw log file._" through the logger which does not appear in your console.

So perhaps you could check the log folder for HDFSLoader.log and see if there is any output in the log file.

Another thing is I did not use the anomaly_label.csv in the code, instead, I used a [label.txt](https://github.com/LeonYang95/PLELog/files/11504662/label.txt) file.

**Comment by xiaooomd** [comment]
Hello, Dr. Yang. I debug the algorithm this afternoon to find out what went wrong. I guess it was caused by the program falling into a loop. 
I wrote two python files in the approaches directory, one named test.py and the other named CONtest.py. 
I mainly carried out the following two steps: 
1 in test.py, I copied the first 15 lines of PLElog.py into the test.py file and added a print statement at the end of the file. 
When I started testing back from the first sentence, I found that when I ran the sentence "from CONSTANTS import *", I got the same result as when I ran PLElog.py. The console did not print any messages, and later found that the program would get stuck in this place as long as it involved importing statements from project. 
The only exception is "from sklearn.decomposition import FastICA". If only this sentence is retained, the program outputs normally. 
2 I created a new CONtest.py file in the approaches directory, copied 1-60 lines of CONSTANTS.py into CONtest.py, ready to test whether there is a problem with a statement in CONSTANTS.py, added a print statement at the end of CONtest.py, commented out "from CONSTANTS import *" in test.py and added a new "from CONtest import *", in order to verify whether the problem is caused by a different directory, the result console does not print any messages. 
I interrupted "if not os.path.exists (log_file_root):" in the GET_LOGS_ROOT (): function, tested it in debug mode, and found that the program was trapped in a dead loop within this function. 
If. 
So I understand that the reason why my console did not print any statements is that the program fell into a loop when guiding the package and was unable to guide the package all the time. The "main" function never ran. I don't know if this is a bug. I hope you can troubleshoot this problem if you are free.
![test](https://github.com/LeonYang95/PLELog/assets/51357799/5597ad14-52da-4a4b-bf79-1875da3704a5)
Only keep "from sklearn.decomposition import FastICA" running
![test1](https://github.com/LeonYang95/PLELog/assets/51357799/7775217b-0af8-4bae-a3f0-83056debde35)
CONtest.py 
![test2](https://github.com/LeonYang95/PLELog/assets/51357799/41203a5a-ae29-4829-b55b-355d5089fc9e)
running with “from CONtest import *”，no result，no print
![test3](https://github.com/LeonYang95/PLELog/assets/51357799/209ca5e2-d65e-47d4-aa4d-ee9ba1d3ded2)
#  LOG_ROOT = GET_LOGS_ROOT()
![test4](https://github.com/LeonYang95/PLELog/assets/51357799/a7920e91-5df9-4d86-97af-4158f252c8bf)
result with “#  LOG_ROOT = GET_LOGS_ROOT()”，call the function normally and output the print statement

**Comment by LeonYang95** [comment]
Thanks a  lot!

The origin design of the ROOTs were to avoid incorrect start position which may affect the relative paths in the code.
 For example, if you run PLELog.py under the root of PLELog or under the approach folder. the relative path will start from different points. Therefore, I have changed the file-related paths to make them start from the PROJECT_ROOT.

So the issue is, I have set the terminate condition of GRT_PROJECT_ROOT method in CONSTANTS.py as if it founds a folder named "PLELog". That is why you were in a dead loop since yours is "PLELog-main". I suppose you downloaded the project by downloading zip instead of cloning.

Anyway, this is an important issue,  and I will fix it, as I have already updated the structure in another project of log-based anomaly detection. Sorry for the inconvenience and thank you so much!

**Comment by xiaooomd** [comment]
thanks a lot！I will continue to try to run the code and look forward to communicating with you

**Comment by superzeroT** [comment]
How can we solve this problem of the endless cycle?

**Comment by LeonYang95** [comment]
For now, please make sure that the root of PLELog is named as 'PLELog'.

The [code](https://github.com/LeonYang95/PLELog/blob/c8bb56b08fe6368f3b3c71ff88de8a87c48c7607/CONSTANTS.py#L43C55-L43C55) will stop until it finds a folder named 'PLELog'.

**Comment by superzeroT** [comment]
the root of PLELog is named as 'PLELog'，But it's still an endless loop.

**Comment by superzeroT** [comment]
The project I cloned

**Comment by LeonYang95** [comment]
@superzeroT Could you please open an new issue and attach your runtime logs so that I can have a more comprehensive understanding of your situation? Thanks a lot.

**Comment by superzeroT** [comment]
This problem is solved.Thank you.
My remote server file root is incorrectly named.
